server/api/tautulli.ts

```diff
@@ -192,23 +193,47 @@ class TautulliAPI {
   public async getUserWatchHistory(
     user: User
   ): Promise<TautulliHistoryRecord[]> {
+    let records: TautulliHistoryRecord[] = [];
+
     try {
       if (!user.plexId) {
         throw new Error('User does not have an associated Plex ID');
       }
 
-      return (
-        await this.axios.get<TautulliHistoryResponse>('/api/v2', {
-          params: {
-            cmd: 'get_history',
-            grouping: 1,
-            order_column: 'date',
-            order_dir: 'desc',
-            user_id: user.plexId,
-            length: 100,
-          },
-        })
-      ).data.response.data.data;
+      const take = 100;
+      let start = 0;
+
+      while (records.length < 20) {
```

Comment:
What happens here when there are not at least 20 records in the first place?

Reply:
It will stop after the first API call that doesn't return any records (see the `if` clause below).

Reply:
Oh, I see. Missed that. 👍